src/main.rs

```diff
@@ -233,7 +258,7 @@ async fn post(path: &str, auth: Option<BearerAuth>) -> Result<HttpResponse> {
         .uri(path)
         .method("POST")
         .header("Host", "localhost")
-        .body(Empty::<Bytes>::new())
+        .body(body.unwrap_or(String::new()))
```

Comment:
nit: this will probably trigger cargo clippy as you create a temporary that is only used if body is None; furthermore `String` implements the Default trait, so you can use `body.unwrap_or_default()`.